### PR TITLE
Quieted logrotate script

### PR DESCRIPTION
**Pull request by nmaludy**
This is a fix for https://github.com/StackStorm/st2/issues/3633

The following updates have been made:

1. Redirect `stdout` to `/dev/null` for every `st2ctl reopen-log-files` command
2. Added `missingok` to `/var/log/st2/st2web.log` because this file does not seem to be created (at least on my Red Hat test box)

### CODE REVIEW

**Comment by nmaludy** [comment]
I just tried booting up an st2vagrant instance and `/var/log/st2/st2web.log` is missing there also.

Is this a legacy log file that needs to be removed from the rotate script?

**Comment by Kami** [comment]
Hm, is this a desired behavior?

In case log rotation fails, I imagine service could print some useful info to stdout (in addition to stderr) which could help with troubleshooting things.

**Comment by Kami** [comment]
As far as st2web goes - Yeah i believe it's an old service since now st2web files are served directly by nginx.

Need to check default nginx config to see where st2web vhost stuff is logged.

**Comment by nmaludy** [comment]
Normally we direct any "standard" information to log files and then any errors get raised so an email is generated.

Working in an environment right now where we send all emails generated by `root` user on linux hosts to a sysadmin distribution list. Any cronjob that writes to `stdout` generates an email and spams the whole DL. In our environment we have about several StackStorm VMs, each sending us a nightly email with the content shown in the issue linked in the original post. This spam behavior is not desired, especially since there are no errors and this is simply rotating logs. We would want to be notified however if the rotation fails and we need to take some sort of action (example, maybe the service dies when logs are rotated).

**Comment by Kami** [comment]
@nmaludy Yeah, something like that could work :+1:

File: conf/logrotate.conf [hunk 1]
```diff
@@ -115,15 +115,16 @@ notifempty
...
-    st2ctl reopen-log-files st2garbagecollector
+    st2ctl reopen-log-files st2garbagecollector > /dev/null
   endscript
 }
 
 ## WebUI
 /var/log/st2/st2web.log {
   daily
   rotate 5
+  missingok
```
Comment (armab) [inline conf/logrotate.conf]: :+1:

File: conf/logrotate.conf [hunk 2]
```diff
@@ -115,15 +115,16 @@ notifempty
   daily
   rotate 5
   postrotate
-    st2ctl reopen-log-files st2garbagecollector
+    st2ctl reopen-log-files st2garbagecollector > /dev/null
   endscript
 }
 
 ## WebUI
 /var/log/st2/st2web.log {
```
Comment (armab) [inline conf/logrotate.conf]: Right, it's outdated.
Please remove entire `st2web` block.

Current `st2web` logs are placed in `/var/log/nginx/` directory: https://github.com/StackStorm/st2/blob/master/conf/nginx/st2.conf#L21-L22 which is logrotated by `nginx`'s rules.

**Comment by nmaludy** [comment]
@armab @Kami Small update based on your comments.

I tried initially to go with @Kami 's approach of redirecting both `stderr` and `stdout` to a file, but also emit `stderr` when `logrotate` was invoked so that an error email was generated. However, this turned out to be a big black hole of redirection and subprocesses invoking `tee`. There ended up being more code in the `postrotate` block to handle log redirections than to restart the service. Also, this code was quite nasty and hard to follow.

I went with @armab 's approach because it was much simpler and straight forward to implement. 

The final change i made was to consolidate the standard and audit logs for a few of the processes. This reduces duplicate code and also allows for a single restart of the service instead of one restart for the standard log rotation and another restart for the audit log rotation.

**Comment by Kami** [comment]
If @armab is fine with just redirecting it to /dev/null, I'm fine with it as well :)

**Comment by Kami** [comment]
As far as "consolidating it" goes - we intentionally had two sections because those are two very different log files.

For the audit ones, users might want to rotate it less often and have different log rotation rules apply to it.

Having said that, if the default values are the same, I'm fine with consolidating it as long as we add a comment and document how to "decouple it" (aka add new section and have different rules apply to audit log files).

**Comment by nmaludy** [comment]
@Kami i've split the configs for the various log files out again. Let me know if you have any other thoughts.

**Comment by Kami** [comment]
Thanks for working on this :+1:

**Comment by Kami** [comment]
Also, to clarify:

> The final change i made was to consolidate the standard and audit logs for a few of the processes. This reduces duplicate code and also allows for a single restart of the service instead of one restart for the standard log rotation and another restart for the audit log rotation.

We don't actually restart a service - we send it `SIGUSR1` signal which causes the service just to re-open the log files without a restart.

**Comment by armab** [comment]
Looks like we reached consensus here.
Thanks everyone, merging :+1:
